### DISCUSSION TITLE
Modification of Turtlebot4 Lite URDF

### DISCUSSION THREAD

**Discussion by YazanRihan** [turtlebot/turtlebot4#245] [Q&A]
Hi all,

Did someone try before to modify the URDF of Turtlbebot4 Lite and get it to work with the rest of its features. I tried making an overlay package to Turtlebot4 description and it worked perfectly in simulation. However, I am not sure how to do it on the real robot as modifying the URDF from /opt/ros/humble/share/turtlebot4_description does not get everything to work well.

**Comment by bmaxdk** [comment]
To address this issue, you can generate an overlay for the `turtlebot4_description` package. This overlay will enable the utilization of your personalized URDF file in place of the preexisting default file. Make sure the overlay is setup correctly.

Also, you need to modify the launch files responsible for initializing the `Turtlebot4` robot to incorporate your tailored URDF. It's probable that you'll need to adjust the file path within these launch files to direct it towards your personalized URDF version within your package. These launch files should be integrated into your workspace, granting you complete authority over their content and configuration.

You may want to adjust your robot configuration

**Comment by slowrunner** [comment]
(Old Post - Still Relevant Question

Is the process:
1) clone entire turtlebot4 to local ROS 2 workspace
2) modify relevant turtlebot4_description urdf files
3) build the workspace
4) run turtlebot4-setup->ROS Setup->Bash Setup
- WORKSPACE_SETUP:  /home/user/ros_ws/build/setup.bash  
- Save
- Robot Upstart
  - Restart
  - Status
- View Settings-->/etc/turtlebot4/setup.bash
  - WORKSPACE_SETUP should have overlay setup.bash
